src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -866,6 +829,10 @@ InferenceProfiler::ValidLatencyMeasurement(
     }
   }
 
+  std::for_each(erase_indices.begin(), erase_indices.end(), [this](size_t i) {
+    this->all_timestamps_.erase(this->all_timestamps_.begin() + i);
```

Comment:
I'm curious how this works when we are removing from the `all_timestamps_` based on index. For example, if the `erase_indices` are `{1, 2, 3}` and the all_timestamps is `{4, 5, 6}`, after the second iteration it is going to change to `{6}` and we want to remove the third element which doesn't exist?

Reply:
Good catch. I switched to a reverse iterator for the erase indices. That will erase things from `all_timestamps_` starting from highest index down to lowest, which should avoid the issue you mentioned 👍